src/rules/for_direction.rs

```diff
@@ -0,0 +1,168 @@
+use std::sync::Arc;
+
+use tree_sitter_lint::{rule, violation, Rule};
+
+pub fn for_direction_rule() -> Arc<dyn Rule> {
+    rule! {
+        name => "for-direction",
+        languages => [Javascript],
+        listeners => [
+            r#"(
+              (for_statement
```

Comment:
I think it's pretty cool how at least for this particular rule so much of the rule "logic" can be expressed in the query itself